src/app/entities/courses/components/course-add-edit-form/course-add-edit-form.component.ts

```diff
@@ -1,59 +1,104 @@
   templateUrl: './course-add-edit-form.component.html',
   styleUrls: [ './course-add-edit-form.component.scss' ],
 })
-export class CourseAddEditFormComponent {
-  title = '';
-  description = '';
-  creationDate = getFormattedCurrentDate();
+export class CourseAddEditFormComponent implements OnInit {
+  courseData: CourseData = null;
+  courseId: number;
 
-  @ViewChild('durationInput') durationInput: DurationInputComponent;
+  constructor(private router: Router, private activatedRoute: ActivatedRoute, private coursesService: CoursesService) { }
 
-  constructor(private router: Router, private coursesService: CoursesService) { }
+  ngOnInit() {
+    this.extractCourseIdFromRouteParams();
+    this.setInitialCourseData();
+    this.handleCourseLoading();
+  }
 
-  get duration() {
-    return this.durationInput.durationValue;
+  get isEditingMode(): boolean {
+    return isValueIntegerNumber(this.courseId);
   }
 
   onSubmit() {
-    this.createCourse();
+    this.submitMethodFactory()();
   }
 
   onCancel() {
     this.navigateToCoursesList();
   }
 
-  private createCourse() {
-    this.coursesService.create({
-      title: this.title,
-      description: this.description,
-      creationDate: new Date(this.creationDate),
-      duration: this.duration,
-      topRated: false,
-    })
-      .pipe(
-        take(1),
-        tap(
-          () => this.navigateToCoursesList(),
-          () => alert(ErrorMessage.CourseCreate),
-        ),
-      )
-      .subscribe()
+  private extractCourseIdFromRouteParams() {
+    this.courseId = parseStringToIntegerNumber(this.activatedRoute.snapshot.params.id);
+  }
+
+  private setInitialCourseData() {
+    this.setCourseData(INITIAL_COURSE_DATA);
+  }
+
+  private handleCourseLoading() {
+    if (this.isEditingMode) {
+      this.getCourseById(this.courseId);
+    }
+  }
+
+  private submitMethodFactory(): () => void {
+    return this.isEditingMode ? this.editCourse.bind(this) : this.createCourse.bind(this);
   }
 
   private navigateToCoursesList() {
     this.router.navigate([ AppRoutePath.Courses ]);
   }
+
+  private setCourseData(courseData: CourseData) {
+    this.courseData = { ...courseData, creationDate: getDatePartFromIsoDateString(courseData.creationDate) };
+  }
+
+  private getCourseById(id: number) {
+    this.coursesService.getById(id).pipe(
+      take(1),
+      tap(
+        course => this.setCourseData(course),
+        () => alert(ErrorMessage.CourseGet),
+      ),
+    ).subscribe();
+  }
+
+  private editCourse() {
+    this.coursesService.update(this.courseId, this.courseData).pipe(
+      take(1),
+      tap(
+        () => this.navigateToCoursesList(),
+        () => alert(ErrorMessage.CourseEdit),
+      ),
+    ).subscribe();
+  }
+
+  private createCourse() {
+    this.coursesService.create(this.courseData).pipe(
+      take(1),
+      tap(
+        () => this.navigateToCoursesList(),
+        () => alert(ErrorMessage.CourseCreate),
+      ),
+    ).subscribe()
```

Comment:
I believe, that `CourseForm` component might be responsible only for editing/creating course data - dummy or data component. Interaction with services and other effects might be performed outside this component.

Reply:
Okay